pkg/tfshim/sdk-v2/internal/rapid/generators.go

```diff
@@ -0,0 +1,167 @@
+package rapidgen
+
+import (
+	"pgregory.net/rapid"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+// schema.Resource docs state this is an abstraction over resources proper, data sources and blocks.
+// This generator creates values representing resources proper.
+func ResourceProperGen(depth int) *rapid.Generator[*schema.Resource] {
+	return rapid.Custom[*schema.Resource](func(t *rapid.T) *schema.Resource {
+		schemaMap := SchemaMapGen(depth).Draw(t, "schemaMap")
+		return &schema.Resource{
+			Schema: schemaMap,
+		}
+	})
+}
+
+// Generates a schema.Resource representing a block.
+func ResourceBlockGen(depth int) *rapid.Generator[*schema.Resource] {
```

Comment:
aren't these two entirely identical?

Reply:
For the moment, but they're distinct logical cases. It's not used yet but I think I'll grow these generators to emulate more features and then this will be a great reminder that Resource blocks and resources proper are different.